docs/resources/scenarios/schedule-interviews-in-teams.md

```diff
@@ -0,0 +1,160 @@
+
+1. An script extracts data from a table and returns an array of objects as JSON data.
+1. The data is then sent to the Teams **Create a Teams meeting** action to send invites.
+1. The same JSON data is sent to another script to update the status of the invitation.
+
+## Scripting skills covered
+
+* Power Automate flows
+* Teams integration
+* Table parsing
+
+## Sample Excel file
+
+Download the file <a href="hr-schedule.xlsx">hr-schedule.xlsx</a> used in this solution and try it out yourself! Be sure to change at least one of the email addresses so that you receive an invite.
+
+## Sample code: Extract table data to schedule invites
+
+Name this script **Schedule Interviews** for the flow.
+
+```TypeScript
+function main(workbook: ExcelScript.Workbook): InterviewInvite[] {
+  const MEETING_DURATION = workbook.getWorksheet("Constants").getRange("B1").getValue() as number;
+  const MESSAGE_TEMPLATE = workbook.getWorksheet("Constants").getRange("B2").getValue() as string;
+
+  // Get the interview candidate information.
+  const sheet = workbook.getWorksheet("Interviews");
+  const table = sheet.getTables()[0];
+  const dataRows = table.getRangeBetweenHeaderAndTotal().getValues();
+
+  // Convert the table rows into InterviewInvite objects for the flow.
+  let invites: InterviewInvite[] = [];
+  dataRows.forEach((row) => {
+    const inviteSent = row[1] as boolean;
+    if (!inviteSent) {
+      const startTime = new Date(Math.round(((row[6] as number) - 25569) * 86400 * 1000));
+      const finishTime = new Date(startTime.getTime() + MEETING_DURATION * 60 * 1000);
+      const candidateName = row[2] as string;
+      const interviewerName = row[4] as string;
+
+      invites.push({
+        ID: row[0] as string,
+        Candidate: candidateName,
+        CandidateEmail: row[3] as string,
+        Interviewer: row[4] as string,
+        InterviewerEmail: row[5] as string,
+        StartTime: startTime.toISOString(),
+        FinishTime: finishTime.toISOString(),
+        Message: generateInviteMessage(MESSAGE_TEMPLATE, candidateName, interviewerName)
+      });
+    }    
+  });
+
+  console.log(JSON.stringify(invites));
+  return invites;
+}
+
+function generateInviteMessage(
+  messageTemplate: string,
+   candidate: string,
+   interviewer: string) : string {
+  return messageTemplate.replace("_Candidate_", candidate).replace("_Interviewer_", interviewer);
+}
+
+// The interview invite information.
+interface InterviewInvite {
+  ID: string
+  Candidate: string
+  CandidateEmail: string
+  Interviewer: string
+  InterviewerEmail: string
+  StartTime: string
+  FinishTime: string
+  Message: string
+}
+```
+
+## Sample code: Mark rows as invited
+
+Name this script **Record Sent Invites** for the flow.
+
+```TypeScript
+function main(workbook: ExcelScript.Workbook, invites: InterviewInvite[]) {
+  const table = workbook.getWorksheet("Interviews").getTables()[0];
+
+  // Get the ID and Invite Sent columns from teh table.
```

Comment:
```suggestion
  // Get the ID and Invite Sent columns from the table.
```

Reply:
It really bothers me that the Docs Authoring Pack doesn't flag "teh" as a spelling error...

Reply:
It thought it was "meh" which is perfectly fine...lol